mne/stats/regression.py

```diff
@@ -392,3 +392,254 @@ def _make_evokeds(coefs, conds, cond_length, tmin_s, tmax_s, info):
             kind='average')  # nave and kind are technically incorrect
         cumul += tmax_ - tmin_
     return evokeds
+
+
+class EncodingModel(object):
+    def __init__(self, delays=None, est=None, preproc_y=True):
+        """Fit a STRF model.
+
+        Fit a receptive field using time lags and a custom estimator or
+        pipeline. This implementation uses Ridge regression and scikit-learn.
+        It creates time lags for the input matrix, then does cross validation
+        to fit a STRF model.
+
+        Parameters
+        ----------
+        delays : array, shape (n_delays,)
+            The delays to include when creating time lags. The input array X
+            will end up having shape (n_feats * n_delays, n_times)
+        est : model estimator | pipeline with estimator
+            The estimator to use for fitting. This is any object that contains
+            a `fit` and `predict` method, which takes inputs of the form
+            (X, y), and which creates a `.coef_` attribute upon fitting.
+            This may be a pipeline, in which case the final estimator must
+            follow the same conventions described above.
+
+        References
+        ----------
+        .. [1] Smith, N. J., & Kutas, M. (2015). Regression-based estimation of
+               ERP waveforms: II. Non-linear effects, overlap correction, and
+               practical considerations. Psychophysiology, 52(2), 169-189.
+        .. [2] Theunissen, F. E. et al. Estimating spatio-temporal receptive
+               fields of auditory and visual neurons from their responses to
+               natural stimuli. Network 12, 289-316 (2001).
+        .. [3] Willmore, B. & Smyth, D. Methods for first-order kernel
+               estimation: simple-cell receptive fields from responses to
+               natural scenes. Network 14, 553-77 (2003).
+        """
+        self.delays = np.array([0]) if delays is None else delays
+        self.n_delays = len(self.delays)
+        self.est = _check_estimator(est)
+        self.preproc_y = preproc_y
+
+    def fit(self, raw, continuous=None, events=None, event_id=None,
```

Comment:
Re: your other thought about taking `Raw` objects to a level above this. Are you imagining that this class would just accept a numpy array? What do you think is the right balance between doing things for the user (e.g., allowing them to pass events indices and auto-creating the unique events columns) vs. giving more non-mne-specific functionality (e.g., allowing numpy arrays and not really doing too much with them before fitting).

Actually, that makes me wonder - perhaps instead of a `delays` parameter, we could treat this as a feature extraction step, and create a `FeatureDelayer` class. Then you could pass that as a part of the pipeline.

...which brings up one more point. I was thinking that rather than passing a sklearn pipeline, you just pass the sklearn estimator, and that we have an extra flag for `pipeline_X` and one for `pipeline_y`, and it would call the `fit_transform` methods of each of these before calling the `fit` method of the estimator. WDYT?